bundles/config/org.eclipse.smarthome.config.discovery/src/main/java/org/eclipse/smarthome/config/discovery/DiscoveryService.java

```diff
@@ -80,6 +80,25 @@
     void startScan(ScanListener listener);
 
     /**
+     * Triggers this service to start an active scan for new devices.<br>
+     * This method must not block any calls such as {@link #abortScan()} and
+     * must return fast.
+     * <p>
+     * If started, any registered {@link DiscoveryListener} must be notified about {@link DiscoveryResult}s.
+     * <p>
+     * If there is already a scan running, it is aborted and a new scan is triggered.
+     * <p>
+     * In contrast to {@link #startScan(ScanListener)} this method can be given a starting time stamp
```

Comment:
This method does not look meaningful on the public facing API (besides, it might even break compatibility to existing services that might not extend the abstract implementation).
It very much looks like a workaround to some problem and it does not even ask the implementation to mandatorily make use of this timestamp in case removeOlderResults() is called.

Isn't there any way to deal with the situation in the DiscoveryServiceRegistry itself? And is it at all legal that a DiscoveryService removes ALL results, even it they were not created by itself...?